Error in option to update add-ons automatically

### Brief summary

When NVDA is configured to update add-ons without notification, add-ons aren't updated.

### Steps to reproduce

1. Go to NVDA`s menu, Preferences submenu, Settings, Add-on store, and form Automatic updates, ensure that Update automatically is the active option.
2. Restart NVDA.
3. Wait until the message informing that add-ons are been updated is shown.

### Actual behavior

- NVDA reports that add-ons are been updated.
- A message asking to restart NVDA is not presented, and add-ons aren't updated.

The log shows the current information:

ERROR - logHandler._threadExceptHook (21:19:00.284) - AutomaticAddonUpdate (436):
Exception in thread AutomaticAddonUpdate:
Traceback (most recent call last):
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "threading.pyc", line 982, in run
  File "gui\message.pyc", line 65, in funcWrapper
  File "gui\addonStoreGui\controls\messageDialogs.pyc", line 613, in _updateAddons
  File "ui.pyc", line 250, in message
  File "braille.pyc", line 2909, in message
  File "braille.pyc", line 2923, in _resetMessageTimer
  File "wx\core.pyc", line 3471, in __init__
  File "wx\core.pyc", line 3492, in Start
wx._core.wxAssertionError: C++ assertion "wxThread::IsMain()" failed at ..\..\src\common\timerimpl.cpp(57) in wxTimerImpl::Start(): timer can only be started from the main thread
INFO - config.ConfigManager._loadConfig (21:19:35.799) - MainThread (15288):
Loading config: C:\Users\Usuariopc\AppData\Roaming\nvda\profiles\mintty.ini
INFO - config.ConfigManager._loadConfig (21:19:35.984) - MainThread (15288):
Loading config: C:\Users\Usuariopc\AppData\Roaming\nvda\profiles\Verbalizar todo.ini


### Expected behavior

A message to restart NVDA should be presented, and add-ons should be updated

### NVDA logs, crash dumps and other attachments

[nvda.log](https://github.com/user-attachments/files/22476194/nvda.log)


### NVDA type

installed copy

### NVDA version

2025.3rc1 (2025.3.0.52513)

### Have you tried any other versions of NVDA? If so, please report their behaviors.

I've tried recent versions from source, with the same behavior.

### Windows version

Versión	10.0.26100 compilación 26100

### Name and version of other software in use when reproducing the issue

Focus 40 braille display.

### Other information about your system

N/A

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have run the COM Registration Fixing Tool and the issue still occurs

--- a/source/addonHandler/__init__.py
+++ b/source/addonHandler/__init__.py
@@ -801,7 +801,7 @@ class Addon(AddonBase):
 		self._importedAddonModules.clear()
 		for modName in set(sys.modules.keys()) - self._modulesBeforeInstall:
 			module = sys.modules[modName]
-			if module.__file__ and module.__file__.startswith(self.path):
+			if module.__name__ and module.__name__.startswith(self.path):
 				log.debug(f"Removing module {module} from cache of imported modules")
 				del sys.modules[modName]
 

--- a/source/gui/addonStoreGui/controls/messageDialogs.py
+++ b/source/gui/addonStoreGui/controls/messageDialogs.py
@@ -590,6 +590,9 @@ class UpdatableAddonsDialog(
 				wx.CallAfter(delayCreateDialog)
 
 			case AddonsAutomaticUpdate.UPDATE:
+
+				# Translators: Message shown when updating add-ons automatically
+				wx.CallAfter(ui.message, pgettext("addonStore", "Updating add-ons..."), SpeechPriority.NEXT)
 				threading.Thread(
 					name="AutomaticAddonUpdate",
 					target=_updateAddons,
@@ -608,9 +611,6 @@ def _updateAddons(addonsPendingUpdate: list[_AddonGUIModel]):
 	This function is treated as message box to prevent NVDA from exiting while the download/install is in progress.
 	"""
 	from ..viewModels.store import AddonStoreVM
-
-	# Translators: Message shown when updating add-ons automatically
-	ui.message(pgettext("addonStore", "Updating add-ons..."), SpeechPriority.NEXT)
 	listVMs = {AddonListItemVM(a, status=getStatus(a, _StatusFilterKey.UPDATE)) for a in addonsPendingUpdate}
 	AddonStoreVM.getAddons(
 		listVMs,

--- a/source/gui/addonStoreGui/controls/messageDialogs.py
+++ b/source/gui/addonStoreGui/controls/messageDialogs.py
@@ -590,7 +590,6 @@ class UpdatableAddonsDialog(
 				wx.CallAfter(delayCreateDialog)
 
 			case AddonsAutomaticUpdate.UPDATE:
-
 				# Translators: Message shown when updating add-ons automatically
 				wx.CallAfter(ui.message, pgettext("addonStore", "Updating add-ons..."), SpeechPriority.NEXT)
 				threading.Thread(
@@ -611,6 +610,7 @@ def _updateAddons(addonsPendingUpdate: list[_AddonGUIModel]):
 	This function is treated as message box to prevent NVDA from exiting while the download/install is in progress.
 	"""
 	from ..viewModels.store import AddonStoreVM
+
 	listVMs = {AddonListItemVM(a, status=getStatus(a, _StatusFilterKey.UPDATE)) for a in addonsPendingUpdate}
 	AddonStoreVM.getAddons(
 		listVMs,

--- a/user_docs/en/changes.md
+++ b/user_docs/en/changes.md
@@ -39,7 +39,7 @@ We recommend using Windows 11, or if that is not possible, the latest Windows 10
 * When reporting the location of the caret in classic versions of Notepad and other Win32 edit controls, text position is now more accurate. (#18767, @LeonarddeR)
 * Fixed a problem where some SAPI 4 voices (e.g. IBM ViaVoice) start speaking in the maximum volume instead of the current volume when speaking capital letters. (#18866, @gexgd0419)
 * NVDA no longer fails to read the contents of wx Web View controls. (#17273, @LeonarddeR)
-* When NVDA is configured to update add-ons automatically without notifying about updates, add-ons can be properly updated. (#18965, @nvdaes)
+* When NVDA is configured to update add-ons automatically in the background, add-ons can be properly updated. (#18965, @nvdaes)
 
 ### Changes for Developers
 

--- a/user_docs/en/changes.md
+++ b/user_docs/en/changes.md
@@ -39,6 +39,7 @@ We recommend using Windows 11, or if that is not possible, the latest Windows 10
 * When reporting the location of the caret in classic versions of Notepad and other Win32 edit controls, text position is now more accurate. (#18767, @LeonarddeR)
 * Fixed a problem where some SAPI 4 voices (e.g. IBM ViaVoice) start speaking in the maximum volume instead of the current volume when speaking capital letters. (#18866, @gexgd0419)
 * NVDA no longer fails to read the contents of wx Web View controls. (#17273, @LeonarddeR)
+* When NVDA is configured to update add-ons automatically without notifying about updates, add-ons can be properly updated. (#18965, @nvdaes)
 
 ### Changes for Developers
 

